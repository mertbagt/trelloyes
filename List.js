import React from 'react';
import Card from './Card'
import './List.css';

function List(props) {   
  return (
    <section className='List'>
      <header className="List-header"> 
        <h2 className='List-header'>
          {props.list.header}
        </h2>
      </header> 
      <div className='List-cards'>
        {props.cards.map((card) =>
          <Card
            key={card.id}
            id={card.id}
            card={card}
            onDeleteCard={props.onDeleteCard}
          />
        )}
        <button 
          onClick={() => props.onRandomCard(props.id)}
          type="button"
          className="List-add-button">
          + Add Random Card
        </button>
      </div>
    </section>
  )
}

List.defaultProps = {
  onClickAdd: () => {},
}

export default List;