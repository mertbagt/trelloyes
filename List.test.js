import React from 'react';
import ReactDOM from 'react-dom';
import List from './List';
import STORE from './STORE';
import renderer from 'react-test-renderer';

describe('List component', () => {
  
  it('renders without crashing', () => {
    const div = document.createElement('div');
    const testCards = [
        { id: 'a', title: 'First card', content: 'lorem ipsum' },
        { id: 'b', title: 'Second card', content: 'lorem ipsum' },
        { id: 'e', title: 'Fifth card', content: 'lorem ipsum' },
        { id: 'f', title: 'Sixth card', content: 'lorem ipsum' },
        { id: 'g', title: 'Seventh card', content: 'lorem ipsum' },
        { id: 'j', title: 'Tenth card', content: 'lorem ipsum' },
        { id: 'l', title: 'Twelfth card', content: 'lorem ipsum' },
        { id: 'm', title: 'Thirteenth card', content: 'lorem ipsum' },
      ];

    ReactDOM.render(
      <List 
        key='1'
        header='First list'
        cards={testCards}
      />,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {  
  const testIds = [ 'a', 'b', 'e', 'f', 'g', 'j', 'l', 'm' ]

  const testCards = [
    { id: 'a', title: 'First card', content: 'lorem ipsum' },
    { id: 'b', title: 'Second card', content: 'lorem ipsum' },
    { id: 'e', title: 'Fifth card', content: 'lorem ipsum' },
    { id: 'f', title: 'Sixth card', content: 'lorem ipsum' },
    { id: 'g', title: 'Seventh card', content: 'lorem ipsum' },
    { id: 'j', title: 'Tenth card', content: 'lorem ipsum' },
    { id: 'l', title: 'Twelfth card', content: 'lorem ipsum' },
    { id: 'm', title: 'Thirteenth card', content: 'lorem ipsum' },
  ];

  const tree = renderer
    .create(
      <List 
        key='1'
        header='First list'
        cards={testCards}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();  
  }); 

});