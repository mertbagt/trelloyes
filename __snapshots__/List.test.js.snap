// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component renders the UI as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2
      className="List-header"
    >
      First list
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3
        className="Card-title"
      >
        First card
      </h3>
      <p
        className="Card-content"
      >
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3
        className="Card-title"
      >
        Second card
      </h3>
      <p
        className="Card-content"
      >
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3
        className="Card-title"
      >
        Fifth card
      </h3>
      <p
        className="Card-content"
      >
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3
        className="Card-title"
      >
        Sixth card
      </h3>
      <p
        className="Card-content"
      >
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3
        className="Card-title"
      >
        Seventh card
      </h3>
      <p
        className="Card-content"
      >
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3
        className="Card-title"
      >
        Tenth card
      </h3>
      <p
        className="Card-content"
      >
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3
        className="Card-title"
      >
        Twelfth card
      </h3>
      <p
        className="Card-content"
      >
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3
        className="Card-title"
      >
        Thirteenth card
      </h3>
      <p
        className="Card-content"
      >
        lorem ipsum
      </p>
    </div>
    <button
      className="List-add-button"
      type="button"
    >
      + Add Random Card
    </button>
  </div>
</section>
`;
